contracts/crosschain/arbitrum/LibArbitrumL1.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import {IBridge as ArbitrumL1_Bridge} from "../../vendor/arbitrum/IBridge.sol";
+import {IInbox as ArbitrumL1_Inbox} from "../../vendor/arbitrum/IInbox.sol";
+import {IOutbox as ArbitrumL1_Outbox} from "../../vendor/arbitrum/IOutbox.sol";
+
+/**
+ * @dev Primitives for cross-chain aware contracts for
+ * [Arbitrum](https://arbitrum.io/).
+ *
+ * This version should only be used on L1 to process cross-chain messages
+ * originating from L2. For the other side, use {LibArbitrumL2}.
+ */
+library LibArbitrumL1 {
+    /**
+     * @dev Returns whether the current function call is the result of a
+     * cross-chain message relayed by `bridge`.
+     */
+    function isCrossChain(address inbox) internal view returns (bool) {
```

Comment:
Given the bridge, it is not possible to get an inbox. 

We don't need the inbox for receiving, but we'll need it for sending. I added the inbox now, so that if we add message sending, we don't end up in a situation where we need to do a breaking change that replaces an address (the bridge) with another address (the inbox)

Reply:
I'm not convinced by this. The bridge code looks like new inboxes can be deployed and added to the system and I don't know if we should be making the decision to use the inbox here at this point, when we don't do any sending. In fact, the current inbox with the retryable ticket mechanism may not be what we'd want to use for sending, if a simpler inbox with less parameters is developed and deployed later.

Reply:
I don't really care about the current inbox interface. What I want to avoid, is a situation where we would do the following update:

from
```
// Old version that uses the bridge to receive
abstract contract CrossChainEnabledArbitrumL1 is CrossChainEnabled {
    ...
    constructor(address bridge) {
        _bridge = bridge;
    }
}
```
to
```
// A new version that is capable of receive AND send
abstract contract CrossChainEnabledArbitrumL1 is CrossChainEnabled {
    ...
    constructor(address inbox) {
        _inbox = inbox;
    }
}
```

Now, if in the future, knowing the bridge is enough to retrieve an inbox, and send a message, then why not...

Reply:
Hm, I'm thinking that new version should rather be:

```
    constructor(address bridge, address inbox) {
```

Reply:
One thing I noticed while changing the code from inbox to bridge is the [documentation](https://developer.offchainlabs.com/docs/public_testnet#l1) includes link to the inbox, but not to the bridge

- Inbox is at [0x578BAde599406A8fE3d24Fd7f7211c0911F5B29e](https://rinkeby.etherscan.io/address/0x578BAde599406A8fE3d24Fd7f7211c0911F5B29e)
- By checking the inbox contract, we can find that the bridge is at [0x9a28e783c47bbeb813f32b861a431d0776681e95](https://rinkeby.etherscan.io/address/0x9a28e783c47bbeb813f32b861a431d0776681e95)
- we can see that the bridge is NOT in the list of important addresses.

If we ask for the bridge, we face a documentation issue.